src/app/features/auth/components/sign-up/sign-up.component.ts

```diff
@@ -0,0 +1,93 @@
+import { Component, OnDestroy, Signal, WritableSignal, computed, effect, inject, signal } from '@angular/core';
+import { FormBuilder, FormControl, FormGroup, FormsModule, ReactiveFormsModule, Validators } from '@angular/forms';
+import { Router, RouterModule } from '@angular/router';
+import { CommonModule } from '@angular/common';
+
+import { catchError, finalize, takeUntil } from 'rxjs/operators';
+import { EMPTY, Subject } from 'rxjs';
+
+import { NzButtonModule } from 'ng-zorro-antd/button';
+import { NzFormModule } from 'ng-zorro-antd/form';
+import { NzGridModule } from 'ng-zorro-antd/grid';
+import { NzInputModule } from 'ng-zorro-antd/input';
+import { NzSpaceModule } from 'ng-zorro-antd/space';
+import { NzTypographyModule } from 'ng-zorro-antd/typography';
+import { NzMessageModule, NzMessageService } from 'ng-zorro-antd/message';
+
+import { AuthService } from '../../../../shared/services/auth.service';
+import { AuthFormValues } from '../../../../shared/interfaces/credentials';
+
+
+
+@Component({
+  selector: 'app-sign-up',
+  standalone: true,
+  imports: [NzFormModule, NzInputModule, NzButtonModule, FormsModule, ReactiveFormsModule, NzGridModule, NzTypographyModule, NzSpaceModule, RouterModule, NzMessageModule, CommonModule],
+  templateUrl: './sign-up.component.html',
+  styleUrl: './sign-up.component.sass'
+})
+export class SignUpComponent implements OnDestroy {
+
+  private router: Router = inject(Router);
+  private authService: AuthService = inject(AuthService);
+  private fb: FormBuilder = inject(FormBuilder);
+  private toast: NzMessageService = inject(NzMessageService)
+
+  private loadingState: WritableSignal<boolean> = signal<boolean>(false)
+
+  private onDestroy$: Subject<void> = new Subject();
+
+  public isLoading: Signal<boolean> = computed(() => this.loadingState())
+
+  public validateForm: FormGroup<AuthFormValues> = this.fb.nonNullable.group({
+    email: ['', [Validators.email, Validators.required]],
+    password: ['', [Validators.minLength(6), Validators.required]],
+  });
+
+  public submitForm(): void {
+    if (this.validateForm.valid) {
```

Comment:
You can also use so-called early return. The code will look like this:

if (this.validateForm.invalid) {
       ....smth
       return;
}

signUp code